bindings/go/src/ucx/listener.go

```diff
@@ -7,24 +7,26 @@ package ucx
 
 // #include <ucp/api/ucp.h>
 import "C"
-import "net"
+import (
+	"net"
+	"unsafe"
+)
 
 type UcpListener struct {
 	listener      C.ucp_listener_h
-	connHandlerId uint64
+	connHandler   unsafe.Pointer
 }
 
 // Needed to call connHandler.Reject() rather than listener.Reject(connHandler)
-var connHandles2Listener = make(map[uint64]C.ucp_listener_h)
+var connHandles2Listener = make(map[unsafe.Pointer]C.ucp_listener_h)
```

Comment:
can we remove it like we removed idToWorker and bundle listener with callback?

Reply:
It's more tricky, because of Go API.
For this we need to have both connection handler and listener in `SetConnectionHandler` but we don't have the listener, it's only created in future call to `NewListener`.
Maybe we can pack a pointer to listener with `connHandler` and dereference it to the listener in `NewListener`.

WDYT?